### PR TITLE
[Refactor] Move _clearing__end_of_subround from trade to messaging

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

File: abce/trade.pyx [hunk 1]
```diff
@@ -659,7 +659,7 @@ class Trade:
             self._reject(offer)
         self._polled_offers = {}
 
-    cdef _reject(self, Offer offer):
+    def _reject(self, Offer offer):
```
Comment (rht) [inline abce/trade.pyx]: I have to rename this to `def` because otherwise cython 0.27.3 won't compile the trade.pyx file.

Comment (DavoudTaghawiNejad) [reply abce/trade.pyx]: I can not reproduce this problem and it comes at a mayor speed penalty. We should investigate this better.

**Comment by DavoudTaghawiNejad** [comment]
I would like to do a refactor that changes the "_x" thing against enums.

File: abce/trade.pyx (hunk 1 above)
Comment (rht) [reply abce/trade.pyx]: Yeah removed.

File: abce/messaging.py [hunk 2]
```diff
@@ -147,3 +147,54 @@ def get_messages_all(self):
             ret[key] = messages
         self._msgs.clear()
         return ret
+
+    def _clearing__end_of_subround(self, incomming_messages):
```
[the remark this answers is not in the document]
Comment (rht) [reply abce/messaging.py]: Renamed to _do_message_clearing

**Comment by rht** [comment]
@DavoudTaghawiNejad I still can't rebuild trade.c because of the cdef issue. Maybe you should take over this PR.

**Comment by DavoudTaghawiNejad** [comment]
@rht can you now install abce?

**Comment by rht** [comment]
Wait, did you merge this PR? The trade.c still uses the trade.pyx that uses `def` instead of `cdef`.

**Pull request merged**